cli/tests/environment-pull.bats

```diff
@@ -181,3 +215,47 @@ function update_dummy_env() {
   run "$FLOX_BIN" pull --remote floxtest/default
   assert_success
 }
+
+# ---------------------------------------------------------------------------- #
+
+# bats test_tags=pull:unsupported
+# pulling an environment without packages for the current platform
+#should fail with an error
+@test "l?: pull environment without packages for the current platform fails" {
+  make_dummy_env "owner" "name"
+  update_dummy_env "owner" "name"
+  make_incompatible "owner" "name"
+
+  run "$FLOX_BIN" pull --remote owner/name
+  assert_failure
+  assert_output --partial "This environment is not yet compatible with your system ($NIX_SYSTEM)"
+
+  run "$FLOX_BIN" pull --remote owner/name --add-system
+  assert_success
+}
+
+# bats test_tags=pull:unsupported:prompt-fail
+# pulling an environment without packages for the current platform
+#should fail with an error
+@test "l?: pull environment without packages for the current platform prompts for about adding system" {
+  update_dummy_env "owner" "name"
+  make_incompatible "owner" "name"
+
+  run -0 expect -d "$TESTS_DIR/pull/promptAmendSystem.exp" owner/name "$NIX_SYSTEM" no
+  assert_success
+  assert_output --partial "This environment is not yet compatible with your system ($NIX_SYSTEM)"
+}
+
+# bats test_tags=pull:unsupported:prompt-success
+# pulling an environment without packages for the current platform
+#should fail with an error
+@test "l?: pull environment without packages for the current platform prompts for about adding system: produces env" {
+  update_dummy_env "owner" "name"
+  make_incompatible "owner" "name"
+
+  run -0 expect -d "$TESTS_DIR/pull/promptAmendSystem.exp" owner/name "$NIX_SYSTEM" yes
+  assert_success
+
+  run "$FLOX_BIN" list
+  assert_success
+}
```

Comment:
suggestion: test both the failed to build after amending system and build succeeded cases